google-analytics-chart.html

```diff
@@ -92,253 +123,301 @@
-          /**
-           * True after the chart has been rendered for the first time.
-           *
-           * @attribute rendered
-           * @type boolean
-           */
-          rendered: {
-            value: false,
-            reflect: true
-          },
-          /**
-           * True if the chart is currently loading data.
-           *
-           * @attribute loading
-           * @type boolean
-           */
-          loading: {
-            value: false,
-            reflect: true
-          }
+        /**
+         * True after the chart has been rendered for the first time.
+         *
+         * @attribute rendered
+         * @type boolean
+         */
+        rendered: {
+          type: Boolean,
+          value: false,
+          notify: true,
+          reflectToAttribute: true
+        },
+        /**
+         * True if the chart is currently loading data.
+         *
+         * @attribute loading
+         * @type boolean
+         */
+        loading: {
+          type: Boolean,
+          value: false,
+          notify: true,
+          reflectToAttribute: true
+        },
+        /**
+         * True if setup is ready
+         *
+         * @attribute setupReady
+         * @type Boolean
+         */
+        setupReady: {
+          type: Boolean,
+          observer: 'setupReadyChanged'
         },
 
-        ready: function() {
-          merge(this.$.chart.options, getChartOptions(this.type), this.options);
+        /**
+         * google-analytics-query passthrough properties
+         * See google-analytics-query for documentation
+         * startDate, endDate, data, ids, metrics, dimensions, sort, filters, segment, samplingLevel, startIndex, maxResults, output, fields
+         */
+        startDate: {
+          type: String,
+        },
+        endDate: {
+          type: String
+        },
+        data: {
+          type: Object
+        },
+        ids: {
+          type: String
+        },
+        metrics: {
+          type: String
+        },
+        dimensions: {
+          type: String
+        },
+        sort: {
+          type: String
+        },
+        filters: {
+          type: String
         },
+        segment: {
+          type: String
+        },
+        samplingLevel: {
+          type: String
+        },
+        startIndex: {
+          type: Number
+        },
+        maxResults: {
+          type: Number
+        },
+        output: {
+          type: String,
+          value: 'dataTable'
+        },
+        fields: {
+          type: String
+        }
+      },
 
-        userAuthorized: function() {
-          this.super();
+      ready: function() {
+        this._boundResponseHandler = this.handleResponse.bind(this);
+        merge(this.$.chart.options, getChartOptions(this.type), this.options);
+      },
 
-          // Get metadata as soon as possible so we don't have to wait later.
+      setupReadyChanged: function(newVal, oldVal) {
+        if (newVal)
```

Comment:
It was just a stylistic thing. All other single line conditional in this repo have the condition statement on the same line as the if.